crates/rpc/src/v03/method/estimate_fee.rs

```diff
@@ -0,0 +1,409 @@
+use crate::{
+    cairo::ext_py::{BlockHashNumberOrLatest, GasPriceSource},
+    context::RpcContext,
+    v02::types::{reply::FeeEstimate, request::BroadcastedTransaction},
+};
+use pathfinder_common::{BlockId, StarknetBlockTimestamp};
+use starknet_gateway_types::pending::PendingData;
+use std::sync::Arc;
+
+#[derive(serde::Deserialize, Debug, PartialEq, Eq)]
+pub struct EstimateFeeInput {
+    request: Vec<BroadcastedTransaction>,
+    block_id: BlockId,
+}
+
+crate::error::generate_rpc_error_subset!(
+    EstimateFeeError: BlockNotFound,
+    ContractNotFound,
+    ContractError,
+    InvalidMessageSelector,
+    InvalidCallData
+);
+
+impl From<crate::cairo::ext_py::CallFailure> for EstimateFeeError {
+    fn from(c: crate::cairo::ext_py::CallFailure) -> Self {
+        use crate::cairo::ext_py::CallFailure::*;
+        match c {
+            NoSuchBlock => Self::BlockNotFound,
+            NoSuchContract => Self::ContractNotFound,
+            InvalidEntryPoint => Self::InvalidMessageSelector,
+            ExecutionFailed(e) => Self::Internal(anyhow::anyhow!("Internal error: {}", e)),
+            // Intentionally hide the message under Internal
```

Comment:
Can the comment elaborate on the why? I assume its because these tend to be verbose; or because they're not useful?

Reply:
This is a copy-paste from old estimateFee code. I don't know the original intention, but these internal messages do not seem very useful for end-users anyway (Like "Wrong database version", "Invalid json sent/received", etc.)